tsk/fs/apfs_compat.cpp

```diff
@@ -14,8 +15,27 @@ extern "C" void error_returned(const char* errstr, ...);
 
 static inline const APFSPoolCompat& to_pool(
     const TSK_POOL_INFO* pool_info) noexcept {
-  const auto pool = static_cast<APFSPoolCompat*>(pool_info->impl);
-  return *pool;
+
+    const auto pool = static_cast<APFSPoolCompat*>(pool_info->impl);
+    return *pool;
+}
+
+static inline const APFSPoolCompat& fs_info_to_pool(
+    const TSK_FS_INFO* fs_info) noexcept {
+
+    IMG_POOL_INFO *pool_img = (IMG_POOL_INFO*)fs_info->img_info;
+    return to_pool(pool_img->pool_info);
```

Comment:
We might want to make the cast a static cast for consistency.

Reply:
"invalid type conversion". I'm not sure this will work since IMG_POOL_INFO isn't the same thing as a TSK_IMG_INFO. They just both start with the same structure so we can sort of pretend one is a subclass of the other.